.github/workflows/test_ci_cases.yaml

```diff
@@ -0,0 +1,42 @@
+name: CI tests
+
+on: 
+  push:
+    branches: [ "main" ]
+
+  pull_request:
+    branches: [ "main" ]
+
+jobs: 
+  CI-tests:
+    runs-on: self-hosted
+    steps: 
+
+      - name: Checkout latest enclave-cc source code
+        uses: actions/checkout@v3
+        with:
+          ref: main
+                
+      - name: Install CoCo operator
```

Comment:
Once CI/CD fails in the intermediate step, the environment is not clean up.
When CI/CD fails, we have to log in to the machine to check the environment which can save the work of reproducing

It is recommended to clean up the environment when ci/cd starts to run.

Reply:
It is hard to clean up the environment at the current stage of this project when it failed to install pods. There are too many factors impacting whether the CI can succeed, such as the remaining space of the CI machine, the network quality and some nonexistent directory paths and etc. It can even be stuck when removing the installed pods.
Currently I just clean up the environment of CI machine manually and rerun the Github actions task if the CI failed.
Maybe the best solution is to back up the system and every time run the CI task, recover the system from backup firstly.